aten/src/ATen/native/cpu/FlashAttentionKernel.cpp

```diff
@@ -306,16 +423,136 @@ void cpu_flash_attention(
   accum_t* buf_data = buf.data_ptr<accum_t>();
   scalar_t* buf_reduced_data = is_reduced_type ? buf_reduced.data_ptr<scalar_t>() : nullptr;
 
+  // Buffer to store padding query
+  scalar_t* query_padding_ptr = nullptr;
+  std::unique_ptr<unsigned short[]> query_padding_data;
+  if (!headSize_even && need_pack) {
+    query_padding_data =
+        std::make_unique<unsigned short[]>(num_thread * qSplitSize * eheadSize);
+    query_padding_ptr = reinterpret_cast<scalar_t*>(query_padding_data.get());
+  }
+  // Buffer to store Key and Value after transforms
+  at::Tensor key_t_reorder = at::empty(
+      {batchSize,
+       num_head,
+       eheadSize,
+       rkvSize},
+      c10::CppTypeToScalarType<scalar_t>::value);
+  auto key_reorder_ptr = key_t_reorder.data_ptr<scalar_t>();
```

Comment:
This is only needed by packed case but always allocated here?

Reply:
Thanks for your comments. Revised.